File: DSA/DSA-VIT/BSTs/binary_tree.cpp
#include<bits/stdc++.h>
using namespace std;

class node{
public:
	int data;
	node*right;
	node*left;

	node(int d)
	{
		data=d;
		node*right=NULL;
		node*left=NULL;
	}
};

node* build_tree()
{
	int d;
	cin>>d;
	if(d==-1)
		return NULL;
	node* root=new node(d);
	root->right=build_tree();
	root->left=build_tree();
	return root;
}

void print(node*root)
{
	if(root==NULL)
		return;
	cout<<root->data<<" ->";
	print(root->right);
	print(root->left);
}

int main()
{
	node*root=build_tree();
	print(root);

	return 0;
}

File: DSA.Important-Questions/BinaryTree.java

public class BinaryTree {
    static class Node{
        int data ;
        Node left;
        Node right;

        Node(int data){
            this.data =data;
            this.left = null;
            this.right = null;
        }
    }

    static class BinaryTree{
        static int idx =-1;
        public static Node buildTree(int nodes[]){
             idx++;
             if(node[idx]==-1){
                return null;
             }

             Node newNode = new Node(nodes[idx]);
             newNode.left = builtTree(nodes);
             newNode.right = builtTree(nodes);

             return newNode;
        }
    }
    public static void main(String[] args){
        int nodes[] = {1,2,4,-1,-1,5,-1,-1,3,-1,6,-1,-1}; // given in preorder sequence
        BinaryTree tree = new BinaryTree();
       Node root =  tree.buildTree(nodes);
       System.out.println(root.data);
    }
    private BinaryTree.Node buildTree(int[] nodes) {
        return null;
    }
}
